Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm

```diff
@@ -122,16 +122,16 @@ static FragmentAndResources createFragmentInternal(LocalFrame&, NSAttributedStri
         [excludedElements addObject:@"object"];
 #endif
 
-    NSURL *baseURL = URL::fakeURLWithRelativePart(emptyString());
+    RetainPtr baseURL = URL::fakeURLWithRelativePart(emptyString()).createNSURL();
 
     // The output base URL needs +1 refcount to work around the fact that NSHTMLReader over-releases it.
```

Comment:
Wow, surprised that this is still true so many years later.

Reply:
There is no radar number in the comment so it is not easy to validate if and when the issue was fixed :/

Reply:
I researched this and found that the over-release still does exist in `NSHTMLReader`. I also found the change that introduced the use of `@"OutputBaseURL"`, which was https://bugs.webkit.org/show_bug.cgi?id=176123 and https://github.com/WebKit/WebKit/commit/eca06b9e1181e7aaae08d34f891bc6dcc0b2afad.

Reply:
It’s possible we need not treat this additional release as a bug but rather as the API contract.

Reply:
rdar://148772051